src/components/Table/TransactionTable.js

```diff
@@ -0,0 +1,213 @@
+   * useful for setting custom TxManager options
+   *
+   * @static
+   * @returns {void}
+   */
+  static setTxManager(txManager) {
+    assert(txManager instanceof TxManager);
+    this.txManager = txManager;
+  }
+
+  /*
+   * Format table data from output of UXTX.toJSON
+   * Maps props.headerMap values that correspond to
+   * UXTX.toJSON values into an object with keys
+   * of the headerMap key
+   *
+   * This allows for arbitrarily selecting
+   * column headers
+   *
+   * @param {Object[]} transactions - list of transaction json
+   * @param {String} wallet - wallet the txs belong to
+   * @returns {Tuple[Object[], Object[]]}
+   *
+   * TODO: allow function in headerMap
+   * that can return a component?
+   */
+  formatTableData(transactions, UXTXOptions) {
+    const txns = this.txManager.parse(transactions, UXTXOptions);
+    const { headerMap, expandedMap } = this.props;
+
+    // tableInput and expandedData are lists
+    // each index corresponds to a single data point
+    // tableInput renders the columns
+    // expandedData renders the dropdown
+    // expandedData mainData renders at the top
+    // expandedData subData renders at the bottom
+    // ugly but functional...
+    const { tableInput, expandedData } = txns.reduce(
+      (acc, tx) => {
+        let tmp = {};
+        for (let [key, val] of Object.entries(headerMap)) tmp[val] = tx[key];
+        acc.tableInput.push(tmp);
+        tmp = { mainData: {}, subData: {} };
+        for (let [key, val] of Object.entries(expandedMap.mainData))
+          tmp.mainData[val] = tx[key];
+        for (let [key, val] of Object.entries(expandedMap.subData))
+          tmp.subData[val] = tx[key];
+        acc.expandedData.push(tmp);
+        return acc;
+      },
+      { tableInput: [], expandedData: [] }
+    );
+
+    return [tableInput, expandedData];
+  }
+
+  /*
+   * Get header values
+   * TODO: cache result and only
+   * compute if things changed
+   *
+   * @returns {String[]}
+   */
+  getHeaders() {
+    return Object.values(this.props.headerMap);
+  }
+
+  /*
+   * Render
+   * @returns {JSX}
+   */
+  render() {
+    const { transactions, UXTXOptions, wallet, protocol } = this.props;
+    if (wallet) UXTXOptions.wallet = wallet;
+    if (protocol) UXTXOptions.protocol = protocol;
+
+    const [tableData, expandedData] = this.formatTableData(
+      transactions,
+      UXTXOptions
+    );
+
+    // TODO: only invoke if headerMap is different
+    const headers = this.getHeaders();
+
+    // TODO: can colHeaders be more than just text?
+    // TODO: make expandedHeight slightly larger if
+    // rendering extra links
+    const { ExpandedComponent, expandedHeight, children } = this.props;
+
+    return (
+      <div>
+        <Table
+          colHeaders={headers}
+          tableData={tableData}
+          expandedHeight={expandedHeight}
+          ExpandedComponent={props => (
+            <ExpandedComponent {...props}>{children}</ExpandedComponent>
```

Comment:
Right now I have it set up so that the children of `Table` are rendered inside of the `ExpandedComponent` at the bottom. From a UX perspective, a dev will want to import the table and pass it props and have it work. It feels a little strange that the children of the `Table` will end up rendering conditionally. Do you think that we should change so that children of `Table` aren't propagated all the way down? Was trying to make it more simple for devs using the component by not needing to do any extra work with the `ExpandedComponent`